frontend/src/components/TopicAddForm.jsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from "react";
+
+const TopicAddForm = ({successCallback, failureCallback}) => {
+    const [newUsername, setNewUsername] = useState("");
+    const [newPassword, setNewPassword] = useState("");
+    const [confirmNewPassword, setConfirmNewPassword] = useState("");
+
+    const onSubmit = async (e) => {
+        e.preventDefault()
+        if(newPassword !== confirmNewPassword){
+            registrationCallback("Passwords do not match!", false)
+        } else {
+            const url = "http://127.0.0.1:5000/comments/"
+            const options = {
+                method: "POST",
+                headers: {
```

Comment:
looks off, because the cr is so long i will go easy and say go check cleaner patterns